server/src/main/java/io/druid/server/namespace/cache/LookupCoordinatorManager.java

```diff
@@ -0,0 +1,571 @@
+  {
+    return ImmutableList.copyOf(
+        Collections2.filter(
+            Collections2.transform(
+                listenerDiscoverer.getNodes(LookupExtractionModule.getTierListenerPath(tier)),
+                HOST_TO_URL
+            ),
+            new Predicate<URL>()
+            {
+              @Override
+              public boolean apply(@Nullable URL input)
+              {
+                return input != null;
+              }
+            }
+        )
+    );
+  }
+
+  static URL getLookupsURL(HostAndPort druidNode) throws MalformedURLException
+  {
+    return new URL(
+        "http",
+        druidNode.getHostText(),
+        druidNode.getPortOrDefault(-1),
+        ListenerResource.BASE_PATH + "/" + LOOKUP_LISTEN_ANNOUNCE_KEY
+    );
+  }
+
+  public void updateLookup(
+      final String tier,
+      final String lookupName,
+      Map<String, Object> spec,
+      final AuditInfo auditInfo
+  )
+  {
+    updateLookups(
+        ImmutableMap.<String, Map<String, Map<String, Object>>>of(tier, ImmutableMap.of(lookupName, spec)),
+        auditInfo
+    );
+  }
+
+  public void updateLookups(final Map<String, Map<String, Map<String, Object>>> spec, AuditInfo auditInfo)
+  {
+    final Map<String, Map<String, Map<String, Object>>> prior = getKnownLookups();
+    if (prior == null && !spec.isEmpty()) {
+      // To prevent accidentally erasing configs if we haven't updated our cache of the values
+      throw new ISE("Not initialized. If this is the first namespace, post an empty map to initialize");
+    }
+    final Map<String, Map<String, Map<String, Object>>> update = prior == null ? spec : new HashMap<>(prior);
+    update.putAll(spec);
+    configManager.set(LOOKUP_CONFIG_KEY, update, auditInfo);
+  }
+
+  public Map<String, Map<String, Map<String, Object>>> getKnownLookups()
+  {
+    if (!started) {
+      throw new ISE("Not started");
+    }
+    return lookupMapConfigRef.get();
+  }
+
+  public boolean deleteLookup(@NotNull final String tier, @NotNull final String lookup, AuditInfo auditInfo)
+  {
+    final Map<String, Map<String, Map<String, Object>>> prior = getKnownLookups();
+    if (prior == null) {
+      LOG.warn("Requested delete lookup [%s]/[%s]. But no lookups exist!", tier, lookup);
+      return false;
+    }
+    final Map<String, Map<String, Map<String, Object>>> update = new HashMap<>(prior);
+    final Map<String, Map<String, Object>> tierMap = update.get(tier);
+    if (tierMap == null) {
+      LOG.warn("Requested delete of lookup [%s]/[%s] but tier does not exist!", tier, lookup);
+      return false;
+    }
+
+    if (!tierMap.containsKey(lookup)) {
+      LOG.warn("Requested delete of lookup [%s]/[%s] but lookup does not exist!", tier, lookup);
+      return false;
+    }
+
+    final Map<String, Map<String, Object>> newTierMap = new HashMap<>(tierMap);
+    newTierMap.remove(lookup);
+    update.put(tier, newTierMap);
+    configManager.set(LOOKUP_CONFIG_KEY, update, auditInfo);
```

Comment:
does this mean we don't care ? if you make this method synchronized we can fix it